services/credential-server-ui/src/pages/ConnectionPage.tsx

```diff
@@ -11,25 +11,23 @@ import { DeleteOutline } from "@mui/icons-material";
 import { UUID_REGEX } from "../constants";
 
 const ConnectionPage: React.FC = () => {
+  const [allContacts, setAllContacts] = useState<Contact[]>([]);
   const [contacts, setContacts] = useState<Contact[]>([]);
-  const [keyword, setKeyword] = useState<string>("");
+  const [connectionsFilter, setConnectionsFilter] = useState<string>("");
 
   useEffect(() => {
     handleGetContacts();
   }, []);
 
   useEffect(() => {
-    handleGetContacts();
-  }, [keyword]);
+    const regex = new RegExp(connectionsFilter, "gi");
+    setContacts(allContacts.filter((contact: { alias: string; id: string }) => regex.test(contact.alias) || regex.test(contact.id)));
```

Comment:
Can't we just do:
```
setContacts(contacts.filter((contact: { alias: string; id: string }) => regex.test(contact.alias) || regex.test(contact.id)));
```

And remove the `allContacts` field

Reply:
If we do that the `contacts` will be filtered everytime we enter a character. When we delete the input it might become an empty array.

Reply:
yep good point